### PR TITLE
fix tiflash test fail

### PR DESCRIPTION
**Pull request by shiyuhang0**
### What problem does this PR solve? 
Some tests will fail for "schema has changed during prewrite!".

image

### Solution
1. Fix IT script for tiflash 6.0 delete the `tiflash_cluster_manager` module https://github.com/PingCAP-QE/ci/pull/901
2. Fix tests by checking TiFlash available before write.

### CODE REVIEW

**Comment by shiyuhang0** [comment]
/run-all-tests test-flash=true tiflash = v5.3.0 tidb = v5.3.0 tikv = v5.3.0 pd = v5.3.0

**Comment by shiyuhang0** [comment]
/run-all-tests test-flash=true tiflash = v5.3.0 tidb = v5.3.0 tikv = v5.3.0 pd = v5.3.0

**Comment by shiyuhang0** [comment]
/run-all-tests test-flash=true tiflash = v5.3.0 tidb = v5.3.0 tikv = v5.3.0 pd = v5.3.0

**Comment by shiyuhang0** [comment]
PTAL @xuanyu66 @birdstorm

File: core/src/test/scala/org/apache/spark/sql/types/pk/MultiColumnPKDataTypeSuites.scala [hunk 1]
```diff
@@ -57,11 +57,11 @@ trait MultiColumnPKDataTypeSuites extends BaseRandomDataTypeTest {
 
   protected val tests: Map[Int, Seq[(Int, Int)]] = {
     val size = dataTypes.size - 1
-    dataTypes.indices
+    val ans = dataTypes.indices
       .flatten { i =>
         dataTypes.indices
           .filter { j =>
-            i != j
+            i < j
```
Comment (xuanyu66) [inline core/src/test/scala/org/apache/spark/sql/types/pk/MultiColumnPKDataTypeSuites.scala:64]: Plz make a description why?

Comment (shiyuhang0) [reply core/src/test/scala/org/apache/spark/sql/types/pk/MultiColumnPKDataTypeSuites.scala:64]: I think the order of the two cols in pk doesn't affect the correct. It's an enhancement that reduces the IT time and does not affect the IT results.
I will add an annotation in the code if you approve this

**Comment by shiyuhang0** [comment]
/run-all-tests test-flash=true tiflash = v5.3.0 tidb = v5.3.0 tikv = v5.3.0 pd = v5.3.0

**Comment by shiyuhang0** [comment]
/run-all-tests test-flash=true tiflash = v5.3.0 tidb = v5.3.0 tikv = v5.3.0 pd = v5.3.0

**Comment by shiyuhang0** [comment]
/run-all-tests test-flash=true tiflash = v5.3.0 tidb = v5.3.0 tikv = v5.3.0 pd = v5.3.0

**Comment by shiyuhang0** [comment]
/run-all-tests test-flash=true tiflash = v5.3.0 tidb = v5.3.0 tikv = v5.3.0 pd = v5.3.0

**Comment by shiyuhang0** [comment]
@xuanyu66 @birdstorm PTAL after the fix

File: core/src/test/scala/org/apache/spark/sql/types/pk/MultiColumnPKDataTypeSuites.scala (hunk 1 above)
Comment (xuanyu66) [reply core/src/test/scala/org/apache/spark/sql/types/pk/MultiColumnPKDataTypeSuites.scala:65]: You can add this in PR description

File: core/src/test/scala/org/apache/spark/sql/types/pk/MultiColumnPKDataTypeSuites.scala [hunk 2]
```diff
@@ -59,6 +59,7 @@ trait MultiColumnPKDataTypeSuites extends BaseRandomDataTypeTest {
     val size = dataTypes.size - 1
     dataTypes.indices
     // filter i<j rather than i!=j to exclude the same tuple (ignore order)
+    // We test if we can fetch the correct data, I think the order of col in pk won't affect it
```
Comment (xuanyu66) [inline core/src/test/scala/org/apache/spark/sql/types/pk/MultiColumnPKDataTypeSuites.scala:62]: It's under the assumption that the order of col in pk wouldn't affect the result of the test.

**Review by xuanyu66 — APPROVED** [review]

File: core/src/test/scala/org/apache/spark/sql/types/pk/MultiColumnPKDataTypeSuites.scala (hunk 1 above)
Comment (birdstorm) [reply core/src/test/scala/org/apache/spark/sql/types/pk/MultiColumnPKDataTypeSuites.scala:65]: I don't thinks so, the order of PK will affect the logic plan.

**Comment by shiyuhang0** [comment]
@birdstorm PTAL

**Review by birdstorm — APPROVED** [review]
LGTM

**Comment by shiyuhang0** [comment]
/run-all-tests test-flash=true tiflash = v5.3.0 tidb = v5.3.0 tikv = v5.3.0 pd = v5.3.0

**Comment by shiyuhang0** [comment]
wait for TiFlash fix their bug

**Pull request merged**